source/extensions/filters/network/kafka/serialization_composite.h

```diff
@@ -0,0 +1,133 @@
+#pragma once
+
+#include <algorithm>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "envoy/buffer/buffer.h"
+#include "envoy/common/exception.h"
+#include "envoy/common/pure.h"
+
+#include "common/common/byte_order.h"
+#include "common/common/fmt.h"
+
+#include "extensions/filters/network/kafka/kafka_types.h"
+#include "extensions/filters/network/kafka/serialization.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace Kafka {
+
+/**
```

Comment:
Why are these templates actually needed? AFAICT all they do is string together multiple `feed` calls. Can't `feed` be an interface method, and then there can be a class that takes a list/vector of feeders and generically does what all of this template code does?

Reply:
I think what you described is basically point `b)` of https://github.com/envoyproxy/envoy/pull/4950#discussion_r234224539

The `feed` part is not a problem, but I'm getting problems with constructing results in `get()`.
If I keep a vector of `Deserializer`-s then their return type is going to have to be something like `void*` or anything sufficiently generic, and then I'd need to convert these results into `Request`'s constructor arguments.

Basically would want to have something like
```
std::vector<Deserializer<?>> delegates_;

ReturnType get() const {
  return { delegates_[0].get(), delegates_[1].get(), delegates_[2].get() .... };
}
```

As a minimum I will `feed` will be changed to be more generic.

When it comes to `get`, I will take a look into possibility of having an array instead of vector - this way I think I'll be able to do some templating to change `array<N>{ deserializer1, ....}` into argument list.

Reply:
OK I see the problem. Alright do what you can and I will take another pass in the next round. Sorry for not seeing the get() issue.